openedx/core/djangoapps/programs/utils.py

```diff
@@ -892,3 +899,45 @@ def _collect_instructors(self, course_run):
             for instructor in course_instructors.get('instructors', []):
                 if instructor.get('name', '').strip() not in curr_instructors_names:
                     self.instructors.append(instructor)
+
+
+def is_user_enrolled_in_program_type(user, program_type):
+    """
+    This method will Look at the learners Enrollments and Entitlements to determine
+    if a learner is enrolled in a Program of the given type.
+
+    NOTE: This method relies on the Program Cache right now. The goal is to move away from this
+    in the future.
+
+    Arguments:
+        user (User): The user we are looking for.
+        program_type (String): The Program type we are looking for.
+
+    Returns:
+        bool: True is the user is enrolled in programs of the requested Type
+    """
+    course_runs = set()
+    course_uuids = set()
+    programs = get_programs_by_type(Site.objects.get_current(), program_type)
+    if not programs:
+        return False
+
+    for program in programs:
+        for course in program.get('courses', []):
+            course_uuids.add(course.get('uuid'))
+            for course_run in course.get('course_runs', []):
+                course_runs.add(course_run['key'])
+
+    # Check Entitlements first, because there will be less Course Entitlements than
+    # Course Run Enrollments.
+    student_entitlements = get_active_entitlement_list_for_user(user)
+    for entitlement in student_entitlements:
+        if str(entitlement.course_uuid) in course_uuids:
+            return True
+
+    student_enrollments = get_enrollments(user.username)
+    for enrollment in student_enrollments:
+        if enrollment['mode'] == 'verified' and enrollment['course_details']['course_id'] in course_runs:
```

Comment:
Just had a late breaking thought about this. Do we still want to be searching just for learners in the `verified` mode here?

I may be getting timelines mixed in my head. I thought we were initially collecting Demographics info from audit and verified MB learners, then possible expanding this to all learners in the future. 

Or, was it Phase 1 -- All verified learners (and learners with Entitlements), Phase 2 -- Paid and Audit MB learners, Phase 3 -- All edX learners?

Reply:
I switched this on the recent version to be all Paid modes.

Reply:
Oh! Good point, I missed that part.

Reply:
Updated to include all modes.